source/shared/components/App.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import settings from 'server/settings';
+import Title from './Title';
+
+const App = ({ dispatch, books }) => {
+  const bookNodes = books.items.map(book => {
+    return (
+      <div key={ book.id }>
+        { book.text } - Read by { book.count } people.
+        <button onClick={ () => dispatch({ type: 'ADD_COUNT', book }) }>Add reader</button>
```

Comment:
Generally using anonymous function as prop is not good since every time render() is called this function is created again and again.
Better "cache" it as instance method or in case of pure component - as function outside of component (if possible) or another pure component.

See https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md for examples and explanation.